dsgrid/query/query_submitter.py

```diff
@@ -32,6 +32,12 @@ def __init__(self, output_dir: Path):
         self._cached_tables_dir().mkdir(exist_ok=True, parents=True)
         self._composite_datasets_dir().mkdir(exist_ok=True, parents=True)
 
+        # TODO: This location will need more consideration.
+        # We might want to store cached datasets in the spark-warehouse and let Spark manage it
+        # for us. However, would we share them on Eagle? What happens on Eagle walltime timeouts
+        # where the tables are left in intermediate states?
```

Comment:
I guess we just have to delete tables if they cannot be reused.

Reply:
Spark does not handle the partial tables, so we would have to catch the specific error and delete those tables in the warehouse.